### PR TITLE
Acceptance testing

### PR DESCRIPTION
**Pull request by t-keazirian**
Trello ticket: https://trello.com/c/PoNVUBzx/43-game-play-through-testing

- Updated README
- Added integration testing for playing the game through
- Created a MockMessage class to use in multiple testing files

Now you can see below there is almost full coverage in `Game`:
Screen Shot 2022-04-27 at 1 55 46 PM

### CODE REVIEW

**Review by tpetersen0308 — APPROVED** [review]
Nice! Do you happen to know if this test library support some sort of test.each behavior? It looks like all of your tests only differ by the input array and expected message, which would be a good candidate for that kind of pattern. Here's an example of how it's done in jest: https://jestjs.io/docs/api#testeachtablename-fn-timeout

File: test/test_acceptance_tests.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+            "1",
+            "1",
+            "2",
+            "3",
+            "5",
+            "4",
+            "8",
+            "n",
+        ],
+    )
```
Comment (Aoxian) [inline test/test_acceptance_tests.py:72]: there is no way to override how this is linting?  .. when linters work against you.

File: test/test_acceptance_tests.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    def test_player_X_wins_the_game(self, mock_input):
+        captured_output = io.StringIO()
+        game = Game(message=MockMessage())
+        expected_message = MockMessage.declare_winner(self, "X")
+
+        sys.stdout = captured_output
+        game.run()
+        sys.stdout = sys.__stdout__
+
+        self.assertIn(expected_message, captured_output.getvalue())
```
Comment (Aoxian) [inline test/test_acceptance_tests.py:20]: there is a pretty cool pattern going on here.  Could you extract a method and remove the duplication?

File: test/test_game.py [hunk 3]
```diff
@@ -3,50 +3,40 @@
...
-class MockMessage:
-    def prompt_for_move(self, current_player):
-        return f"Player {current_player} - prompt for turn"
-
-    def game_over_message(self):
-        return "Game over message"
-
-    def choose_player_symbol(self):
-        return "Player __ Choose your symbol:"
+from test.mocks.mock_message import MockMessage
```
Comment (Aoxian) [inline test/test_game.py:6]: 🎉

File: README.md [hunk 4]
```diff
@@ -17,26 +17,39 @@ This game uses `python 3.1.0`
...
 - The game will go through a series of prompts for a two-player TTT game.
 - You will be notified if you input an incorrect value and be re-prompted to correct the value.
-- The game will declare a winner and gives the option to play again.
+- The game will declare a winner and give the option to play again.
 
-### Testing:
+## Testing:
 
-This project uses [unittest](https://docs.python.org/3/library/unittest.html), a Python testing framework.
+This project uses [unittest][1], a Python testing framework.
```
Comment (Aoxian) [inline README.md:24]: has everything been migrated over to `unittest` or are there still `pytest`s as well?  I may have missed it.

Comment (t-keazirian) [reply README.md:24]: I actually never ended up using `pytest`.  I thought I would but we found our solution with `unittest` instead.  If I end up using it in the future, I can add it back into the README but for now, only `unittest` is being used.

Comment (Aoxian) [reply README.md:24]: ok, thanks.

Comment (t-keazirian) [reply README.md:24]: Did you see I used your README trick for links?!

File: test/test_acceptance_tests.py (hunk 1 above)
Comment (t-keazirian) [reply test/test_acceptance_tests.py:72]: Mmmm...I could check and see but I'm not sure...

File: test/test_acceptance_tests.py (hunk 2 above)
Comment (t-keazirian) [reply test/test_acceptance_tests.py:20]: Hmmmm....it's possible!  gonna need to think on that...

File: README.md (hunk 4 above)
Comment (Aoxian) [reply README.md:24]: I did!  footnotes in markdown are wonderful! 🎉

File: test/test_acceptance_tests.py (hunk 1 above)
Comment (Aoxian) [reply test/test_acceptance_tests.py:72]: don't let it block you; but sometimes, the linters get it wrong.  I think this is one of those times.

File: test/test_acceptance_tests.py (hunk 2 above)
Comment (Aoxian) [reply test/test_acceptance_tests.py:20]: I think @tpetersen0308 have the same [idea](https://github.com/t-keazirian/tic-tac-toe/pull/25#pullrequestreview-955554854)

**Comment by Aoxian** [comment]
I created my own branch, parallel to this one for the PR, and did my own rebase.  I suggest you compare your commits to the ones in mine and note the difference.

**Review by Aoxian — APPROVED** [review]

File: test/test_acceptance_tests.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
+import io
+import sys
+import unittest
+from unittest.mock import patch
+from src.game import Game
+from test.mocks.mock_message import MockMessage
+
+
+class TestAcceptance(unittest.TestCase):
+    def capture_output(self):
```
Comment (t-keazirian) [inline test/test_acceptance_tests.py:10]: @Aoxian something like this for extraction?

Comment (Aoxian) [reply test/test_acceptance_tests.py:10]: yes, though I don't think the name of this function is all that great.

Comment (Aoxian) [reply test/test_acceptance_tests.py:10]: I would have went for something along the lines of `game_playthrough()` and have it return:
```python
    return. captured_output.get_value()
```

Then in your tests:
```python
@patch("builtins.input", side_effect=["1", "1", "1", "2", "3", "4", "5", "6", "7", "n"])
    def test_player_X_wins_the_game(self, mock_input):
    expected_message = MockMessage.declare_winner(self, "X")

    game_output = self.game_playthrough()

    self.assertIn(expected_message, game_output)
```

Comment (t-keazirian) [reply test/test_acceptance_tests.py:10]: Got it.  I had just created a refactor branch so I will do it there.  I like that approach better.

Comment (t-keazirian) [reply test/test_acceptance_tests.py:10]: https://github.com/t-keazirian/tic-tac-toe/pull/26

**Pull request merged**